revdb/src/postgresql_connection.cpp

```diff
@@ -59,11 +61,113 @@ Bool DBConnection_POSTGRESQL::connect(char **args, int numargs)
 	if (t_delimiter != NULL)
 	{
 		t_port = (t_delimiter + (1 * sizeof(char)));
-		*t_delimiter = NULL;
+		*t_delimiter = '\0';
 	}
-	
-	dbconn = NULL;
-	dbconn = PQsetdbLogin(t_host, t_port, NULL, NULL, t_database, t_user, t_password);
+
+    const char *t_sslmode;
+    t_sslmode = NULL;
+    const char *t_sslcompression;
+    t_sslcompression = NULL;
+    const char *t_sslcert;
+    t_sslcert = NULL;
+    const char *t_sslkey;
+    t_sslkey = NULL;
+    const char *t_sslrootcert;
+    t_sslrootcert = NULL;
+    const char *t_sslcrl;
+    t_sslcrl = NULL;
+
+    // extract any ssl options spcified as key value pairs in the final args to revOpenDatabase
+    // e.g. sslmode=require
+    const char *t_ssl_opt_key;
+    t_ssl_opt_key = NULL;
+    const char *t_ssl_opt_val;
+    t_ssl_opt_val = NULL;
+    for (int i = 4; i < numargs; i++)
+    {
+        t_delimiter = strchr(args[i], '=');
+        if (t_delimiter != NULL)
+        {
+            *t_delimiter = '\0';
```

Comment:
You shouldn't really do this - the string is a const char * - it probably shouldn't be mutated ;)